sources/packages/api/src/cra-integration/cra-files/cra-file-header.ts

```diff
@@ -30,4 +31,14 @@ export class CRAFileHeader implements CRAFileLine {
     header.append("0");
     return header.toString();
   }
+
+  public static CreateFromLine(line: string): CRAFileHeader {
+    const header = new CRAFileHeader();
+    header.transactionCode = line.substr(0, 4) as TransactionCodes;
+    header.processDate = dayjs(line.substr(28, 8), DATE_FORMAT).toDate();
+    header.programAreaCode = line.substr(37, 4);
+    header.environmentCode = line.substr(41, 1);
+    header.sequence = parseInt(line.substr(42, 5));
```

Comment:
will it to good to check the character line.substr(42, 5) is a digit before doing parseInt- just in any scenario what if it's not a digit. or to add  a exception handling

Reply:
The files are fixed-sized and there is no scenario where it will not be a number, it is a mandatory field. In case of something really bad happen, like if CRA send a file with bad format, it will throw an error that will be captured in the processing logs.
IMHO we should not add contingencies to the code for a scenario that is not supposed to happen.
Please let me know if you are comfortable with the explanation or if we need further discussion.